tests/integration_tests/validation/validation_util.py

```diff
@@ -0,0 +1,37 @@
+import os
+import pytest
+
+from arelle.CntlrCmdLine import parseAndRun
+from arelle import ModelDocument
+
+
+def get_test_data(args):
+    """
+    Produces a list of Pytest Params that can be fed into a parameterized pytest function
+
+    :param args: The args to be parsed by arelle in order to correctly produce the desired result set
+    :type args: list of strings
+    :return: A list of PyTest Params that can be used to run a parameterized pytest function
+    :rtype: list of ::class:: `~pytest.param`
+    """
+    cntlr = parseAndRun(args)
+    results = []
+    model_document = cntlr.modelManager.modelXbrl.modelDocument
+    if model_document is not None:
+        if model_document.type == ModelDocument.Type.TESTCASESINDEX:
+            for tc in sorted(model_document.referencesDocument.keys(), key=lambda doc: doc.uri):
+                uri_dir_parts = os.path.dirname(tc.uri).split('/')
```

Comment:
Wondering if we can clean this path resolution up a bit. I think you are using strings here instead of internal python file utilities

Reply:
What do you have in mind here?